--- meson.build	2019-01-02 19:21:57.000000000 +0100
+++ meson_new.build	2019-01-27 12:10:35.624634442 +0100
@@ -44,14 +44,14 @@
 executable('pamac-system-daemon',
 	sources: ['common_daemon.vala', 'pamac_config.vala', 'alpm_config.vala', 'alpm_utils.vala', 'system_daemon.vala'],
 	dependencies: [vala_deps, alpm_deps, json, libsoup, polkit, libcurl, appstream],
-	vala_args: [common_vala_args, alpm_vala_args, '--thread'],
+	vala_args: [common_vala_args, alpm_vala_args],
 	c_args: [common_c_args, alpm_c_args],
 	install : true)
 
 libpamac = library('pamac',
 	sources: [common_sources, 'error.vala', 'alpm_config.vala', 'aur.vala', 'database.vala', 'transaction_interface.vala', 'alpm_utils.vala', 'transaction_interface_root.vala', 'transaction_interface_daemon.vala', 'transaction.vala'],
 	dependencies: [alpm_deps, math, json, appstream, libsoup, libcurl],
-	vala_args: [common_vala_args, alpm_vala_args, '--thread'],
+	vala_args: [common_vala_args, alpm_vala_args],
 	c_args: [common_c_args, alpm_c_args],
 	vala_gir: 'Pamac-1.0.gir',
 	install: true,
@@ -112,4 +112,4 @@
 	dependencies: [gtk3, libpamac_dep, libpamac_gtk_dep],
 	vala_args: common_vala_args,
 	c_args: common_c_args,
-	install: true)
+	install: true) 
