diff -Npur "ndiswrapper-1 .59.orig/driver/ndis.c" ndiswrapper-1.59/driver/ndis.c
--- "ndiswrapper-1 .59.orig/driver/ndis.c"	2013-11-28 20:42:36.000000000 +0100
+++ ndiswrapper-1.59/driver/ndis.c	2015-04-13 13:10:02.571539485 +0200
@@ -22,6 +22,10 @@
 #include <asm/dma.h>
 #include "ndis_exports.h"
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0))
+#define strnicmp strncasecmp
+#endif
+
 #define MAX_ALLOCATED_NDIS_PACKETS TX_RING_SIZE
 #define MAX_ALLOCATED_NDIS_BUFFERS TX_RING_SIZE
 
